### ISSUE TITLE
max_runtime ignored for TDL

### ISSUE THREAD

**Issue by htmai-880** [dice-group/Ontolearn#582]
# The issue

The concept learner [TDL](https://github.com/dice-group/Ontolearn/blob/develop/ontolearn/learners/tree_learner.py#L208) has a `max_runtime` argument which is currently ignored.

However, the [paper](https://papers.dice-research.org/2025/ECML_TDL/tdl-public.pdf) mentions in Table 6 that the runtime is limited to 30s.

Would it be possible to implement this without too much effort? I could actually really use some sort of early stopping criterion.

**Comment by Demirrr** [comment]
@TheSheep-stack would you take a look at it?

**Comment by Demirrr** [comment]
@htmai-880 please take a look at the input arguments of tDL. By default, tDL learns expressions involving nominals and more as well. Consequently, I would suggest to specify the DL in which tDL operates

**Comment by htmai-880** [comment]
@Demirrr Afaik, the Mutagenesis ontology includes nominals, right? I was mainly trying to test TDL on available benchmarks before jumping to concrete use cases

**Comment by Demirrr** [comment]
Table 1 of [Tree-Based OWL Class Expression Learner over Large Graphs](https://papers.dice-research.org/2025/ECML_TDL/tdl-public.pdf) suggests that 
tDL cannot learn Restrictions involving nominals (1) and data properties (2).
Yet, after the publication, we continue on improving the tDL's implementation.
With the recent changes, we have the following boolen arguments introduced

https://github.com/dice-group/Ontolearn/blob/5766c78000bdd5a6820d4cb41c6be46e49ac07d4/ontolearn/learners/tree_learner.py#L214

and

https://github.com/dice-group/Ontolearn/blob/5766c78000bdd5a6820d4cb41c6be46e49ac07d4/ontolearn/learners/tree_learner.py#L213
to control (1) and (2). To reproduce the reported results, please set (1) and (2) to False.

Currently, [DecisionTreeClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeClassifier.html
) does not accept an early stopping argument. 
For the time being, the max_runtime criterion cannot be applied for tDL as it is applied in other models.
Instead, one can use `max_depth` argument to find a good balance between the runtime and training accuracy.

Perhaps, we @TheSheep-stack  can use `max_runtime`  in https://github.com/dice-group/Ontolearn/blob/5766c78000bdd5a6820d4cb41c6be46e49ac07d4/ontolearn/learners/tree_learner.py#L595 to throw a Runtime exception if the feature matrix is not created within the allowed time.

**Comment by TheSheep-stack** [comment]
@Demirrr 

Implemented in https://github.com/dice-group/Ontolearn/commit/85a63edd4dc5ec1823f0244cea9f05bbaacadff9

During feature extraction and feature matrix generation the max_runtime is checked and RuntimeException is thrown if overstepped.
When max_runtime is set to None -> no runtime restriction.
